packages/circuits/circuits/incrementalMerkleTree.circom

```diff
@@ -15,35 +15,29 @@ template MerkleTreeInclusionProof(n_levels) {
     signal input path_elements[n_levels];
     signal output root;
 
-    component hashers[n_levels];
-    component mux[n_levels];
-
     signal levelHashes[n_levels + 1];
     levelHashes[0] <== leaf;
 
     // don't allow inclusion proof for 0 leaf
-    component leaf_zero = IsZero();
-    leaf_zero.in <== leaf;
-    leaf_zero.out === 0;
+    signal leaf_zero <== IsZero()(leaf);
+    leaf_zero === 0;
 
+    signal mux[n_levels][2];
+    signal hashers[n_levels];
+    var merklePath[2][2];
     for (var i = 0; i < n_levels; i++) {
         // Should be 0 or 1
         path_index[i] * (1 - path_index[i]) === 0;
 
-        hashers[i] = Poseidon(2);
-        mux[i] = MultiMux1(2);
-
-        mux[i].c[0][0] <== levelHashes[i];
-        mux[i].c[0][1] <== path_elements[i];
-
-        mux[i].c[1][0] <== path_elements[i];
-        mux[i].c[1][1] <== levelHashes[i];
-
-        mux[i].s <== path_index[i];
-        hashers[i].inputs[0]  <== mux[i].out[0];
-        hashers[i].inputs[1]  <== mux[i].out[1];
+        merklePath[0][0] = levelHashes[i];
+        merklePath[0][1] = path_elements[i];
+        merklePath[1][0] = path_elements[i];
+        merklePath[1][1] = levelHashes[i];
+        
```

Comment:
I am thinking about we should assign constraints here
to make sure it works like input signals